### ISSUE TITLE
Is sending multiple requests in one useEtherSWR call the same as batching calls?

### ISSUE THREAD

**Issue by jviray** [aboutlo/ether-swr#17]
Similar to the example in the README I'm making multiple requests in one, but instead of using the same contract, I'm calling functions on different contracts (all with the same abi). Looks something like this.

```  
const ABIs = [
 [address_1, abi],
 [address 2, abi],
 [address 3, abi]
]

const { data: totals, error: errors } = useEtherSWR(
  [
    [address_1, 'getValue'],
    [address_2, 'getValue'],
    [address_3, 'getValue'],
  ],
  { ABIs: new Map(ABIs) }
);
```

Everything returns fine, but when I check the testnet logs, it's still showing 3 separate contract calls being made. Is this the expected behavior?

Maybe i'm misunderstanding the difference between "batching" and/or "multicall", but I'm tying to reduce the load on our Infura node and I want to reduce the amount of contract calls.

**Comment by aboutlo** [comment]
hey @jviray it's correct right now multicall doesn't batch yet. It's syntax sugar but I would like to implement https://github.com/makerdao/multicall.js directly in the [fetcher](https://github.com/aboutlo/ether-swr/blob/master/src/ether-js-fetcher.ts).

**Comment by jviray** [comment]
@aboutlo awesome, I see the TO DO comment in the fetcher file about using https://github.com/Destiner/ethcall.

[useDapp](https://github.com/EthWorks/useDApp) has a useChainCalls hook that uses MakerDao's multicall and also caches each request. Just need a simple, easy to use fetcher like you have here though.

There's also https://github.com/indexed-finance/multicall which enables multi-calls without needing to deploy an aggregator contract

**Comment by sambacha** [comment]
> @aboutlo awesome, I see the TO DO comment in the fetcher file about using https://github.com/Destiner/ethcall.
> 
> 
> 
> [useDapp](https://github.com/EthWorks/useDApp) has a useChainCalls hook that uses MakerDao's multicall and also caches each request. Just need a easy to use fetcher (with multicall + caching) like you have here though, so I'll be looking forward to it.
> 
> 
> 
> There's also https://github.com/indexed-finance/multicall which enables multi-calls without needing to deploy an aggregator contract
> 
> 

You might be interested in the work 1inch has done with chunking requests for multicall, see https://github.com/1inch/multicall

Additionally you might find the Hughes batching system interesting for multicall usage (not the same as makerdaos) https://dl.acm.org/doi/abs/10.1145/3457337.3457839

You can create your own bespoke multicall by creating a contract interface and using that via eth_call. Arbitrium uses a similar technique I believe

**Comment by aboutlo** [comment]
hey @jviray and @sambacha have a look #18

**Comment by aboutlo** [comment]
fixed with #19

**Comment by sambacha** [comment]
Thanks 💯
